src/test/java/org/junitpioneer/jupiter/json/ArrayNodeToListTests.java

```diff
@@ -0,0 +1,216 @@
+					"T. S. Eliot: [The Hollow Men (1925), Ash Wednesday (1930)]");
+	}
+
+	@Test
+	@DisplayName("does not use the converter for non-JSON arguments for simple argument types")
+	void doesNotUseConverterSimple() {
+		ExecutionResults results = PioneerTestKit
+				.executeTestMethodWithParameterTypes(ArrayNodeToListTests.class, "doesNotConvertNonNodesSimple",
+					String.class);
+
+		assertThat(results).hasNumberOfSucceededTests(2);
+		assertThat(results).hasNumberOfReportEntries(2).withValues("12", "23");
+	}
+
+	@Test
+	@DisplayName("does not use the converter for non-JSON arguments for List argument types")
+	void doesNotUseConverterList() {
+		ExecutionResults results = PioneerTestKit
+				.executeTestMethodWithParameterTypes(ArrayNodeToListTests.class, "doesNotConvertNonNodesList",
+					List.class);
+
+		assertThat(results).hasNumberOfSucceededTests(1);
+		assertThat(results).hasNumberOfReportEntries(1).withValues("[a, b, c]");
+	}
+
+	@Test
+	@DisplayName("throws a ParameterResolutionException if it can not convert complex objects")
+	void throwsForMalformedComplexObjects() {
+		ExecutionResults results = PioneerTestKit
+				.executeTestMethodWithParameterTypes(ArrayNodeToListTests.BadConfigurationTestCase.class,
+					"conversionException", List.class);
+
+		assertThat(results)
+				.hasSingleFailedContainer()
+				.withExceptionInstanceOf(UncheckedIOException.class)
+				.hasMessageContaining("Failed to convert to type");
+	}
+
+	@ParameterizedTest
+	@JsonClasspathSource(COMPOSERS)
+	@ReportEntry("{1}")
+	void convertFromClasspath(@Property("name") String name, @Property("music") List<String> works) {
+		then(name).isNotEmpty();
+		then(works).hasSize(2);
+	}
+
+	@ParameterizedTest
+	@JsonSource({ "{ 'single': [1, 4, 7] }", "{ 'single': [2, 4, 9] }" })
+	@ReportEntry("{0}")
+	void convertFromAnnotation(@Property("single") List<Integer> numbers) {
+		then(numbers).hasSize(3);
+	}
+
+	@ParameterizedTest
+	@JsonSource({ "{ 'statements': [true, false, true] }", "{ 'statements': [false, true, false] }" })
+	@ReportEntry("{0}")
+	void convertWithExplicitListType(@Property("statements") LinkedList<Boolean> numbers) {
+		then(numbers).hasSize(3);
+	}
+
+	@ParameterizedTest
+	@JsonClasspathSource(POETS)
+	@ReportEntry("{0}: {1}")
+	void convertToComplexObject(@Property("name") String name, @Property("works") List<Poem> poems) {
+		if (name.equals("T. S. Eliot")) {
+			then(poems).extracting(Poem::getRelease).noneMatch(i -> i < 1900);
+		} else {
+			then(poems).extracting(Poem::getRelease).noneMatch(i -> i > 1900);
+		}
+	}
+
+	@ParameterizedTest
+	@ValueSource(strings = { "12", "23" })
+	@ReportEntry("{0}")
+	void doesNotConvertNonNodesSimple(@Property("nope") String i) {
+		then(i).isNotNull();
+	}
+
+	@ParameterizedTest
+	@MethodSource("letters")
+	@ReportEntry("{0}")
+	void doesNotConvertNonNodesList(@Property("nope") List<String> letters) {
```

Comment:
Is this test needed? It only tests `@ValueSource` and I don't think that this was the intention

Reply:
Not needed. Deleted. Thanks!